src/rust/ide/lib/view/src/graph_editor/component/node/port/output.rs

```diff
@@ -582,43 +611,88 @@ impl OutputPortsData {
 #[derive(Debug,Clone,CloneRef)]
 pub struct OutputPorts {
     /// The FRP api of the `OutPutPorts`.
-    pub frp     : Frp,
-        network : frp::Network,
-        data    : Rc<OutputPortsData>,
+    pub frp              : Frp,
+        network          : frp::Network,
+        network_internal : Rc<RefCell<frp::Network>>,
+        data             : Rc<OutputPortsData>,
+        type_map         : TypeColorMap,
```

Comment:
Hmm, I see this type_map already in second place in the code. Are we keeping the same information twice in different code places or is this a different information?

Reply:
We keep it both in the `OutputPorts` and the `port::Manager` as both of these need to color their ports. I guess we could move this functionality into the node, or alternatively create/refactor a unified `Manager` for both input and output ports.

Reply:
As I've written in discord:

You have 2 functions set_expression_type which do the same, they should be merged together. You have duplicated logic of this type_map – this needs to be refactored in one place. I'm ok with keeping references to a single, internally mutable type_map if you want. Or even keeping immutable type-map (see the im libarary - this could sometimes be useful - a node can create type map and can distribute it to input and output port manager. The im hashmaps are immutable and have fast clone (based on Rc)). Anyway, whatever solution we choose, we should never duplicate the logic of creating a strucutre, so lets create and manage type map in a single place.